recipes/eleuther_eval.py

```diff
@@ -4,40 +4,279 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+import importlib.metadata
 import sys
 import time
 
-from typing import Any, Dict, List, Tuple, Union
+from typing import Dict, List, Tuple, Union
+
+import PIL
 
 import torch
 from omegaconf import DictConfig
 
-from torch import nn
-from torchtune import config, generation, training, utils
-from torchtune.data import left_pad_sequence
+from torchtune import config, training, utils
+from torchtune.data import (
+    format_content_with_images,
+    left_pad_sequence,
+    Message,
+    padded_collate_tiled_images_and_mask,
+)
+from torchtune.generation import generate, sample
 from torchtune.modules import TransformerDecoder
+from torchtune.modules.model_fusion import DeepFusionModel
 from torchtune.modules.tokenizers import ModelTokenizer
+from torchtune.modules.transforms import Transform
 from torchtune.recipe_interfaces import EvalRecipeInterface
 
-logger = utils.get_logger("DEBUG")
-
 try:
-    import lm_eval
-    from lm_eval.evaluator import evaluate, get_task_list
-    from lm_eval.models.huggingface import HFLM
-    from lm_eval.tasks import get_task_dict, TaskManager
-    from lm_eval.utils import make_table
-except ImportError:
-    logger.error(
-        "Recipe requires EleutherAI Eval Harness v0.4. Please install with `pip install lm_eval==0.4.*`"
+    lm_eval_version = importlib.metadata.version("lm_eval")
```

Comment:
Instead of checking via importing legacy functions, we actually check the version. 

This is the correct way lol